lib/livebook_web/controllers/session_controller.ex

```diff
@@ -19,28 +19,42 @@ defmodule LivebookWeb.SessionController do
       {:ok, session} ->
         notebook = Session.get_notebook(session.pid)
 
-        send_notebook_source(conn, notebook, format)
+        file_name =
+          case session.file do
+            nil ->
+              notebook.name
+              |> String.downcase()
+              |> String.replace(" ", "_")
+
+            _ ->
+              %{"file_name" => file_name} =
+                Regex.named_captures(~r{.*[/](?<file_name>.*)[\.].*}, session.file.path)
+
+              file_name
```

Comment:
I think this is equivalent to `file_name |> Path.basename() |> Path.rootname()`?

Reply:
Yup, you're correct. Making changes now.

Reply:
We have a function specific for the our file struct, so `session.file |> Livebook.FileSystem.File.name() |> Path.rootname()` :)